core/types/types.go

```diff
@@ -27,8 +27,9 @@ type Extrinsic []byte
 
 // Block defines a state block
 type Block struct {
-	Header BlockHeader
-	Body   BlockBody
+	Header      BlockHeader
+	Body        BlockBody
+	ArrivalTime uint64 // arrival time of this block
```

Comment:
Please move this to an unexported field in BlockHeader and add a getter for it. That will stop it from being included in any encoding. Lmk if this introduces any issues for you.